src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,129 @@
+                Long id = generatedKeys.getObject(COLUMN_INDEX, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(EXCEPTION_CAN_NOT_ADD_BOOK + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<mate.academy.model.Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(QUERY_FIND_BOOK_BY_ID)) {
+            statement.setLong(PARAMETER_INDEX_ONE, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(EXCEPTION_NOT_FIND_BOOK_BY_ID + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(QUERY_FIND_ALL_BOOKS)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = getBook(resultSet);
+                bookList.add(book);
+            }
+            return bookList;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public mate.academy.model.Book update(mate.academy.model.Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(QUERY_UPDATE_BOOK)) {
+            statement.setString(PARAMETER_INDEX_ONE, book.getTitle());
+            statement.setBigDecimal(PARAMETER_INDEX_TWO, book.getPrice());
+            statement.setLong(PARAMETER_INDEX_THREE, book.getId());
+            if (statement.executeUpdate() < EDGE_VALID_NUMBER_AFFECTED_ROWS) {
+                throw new RuntimeException(EXCEPTION_NO_UPDATE_BOOKS);
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException(EXCEPTION_CAN_NOT_UPDATE_BOOKS + book.getId(), e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(QUERY_DELETE_BOOK_BY_ID)) {
+            statement.setLong(PARAMETER_INDEX_ONE, id);
+            return statement.executeUpdate() > EDGE_INVALID_NUMBER_AFFECTED_ROWS;
+        } catch (SQLException e) {
+            throw new DataProcessingException(EXCEPTION_CAN_NOT_UPDATE_BOOKS + id, e);
+        }
+    }
+
+    private static Book getBook(ResultSet resultSet) throws SQLException {
```

Comment:
I think it should not be static.

Reply:
Let's rename. `getBook` sounds like it is a getter but not a utility method.
`mapResultSetToBook`, `mapToBook`, `extractBook` - examples.